src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1121,18 +1121,16 @@ private void ImportDataMembers()
                             DataMember memberContract = new DataMember(member);
 
                             memberContract.Name = DataContract.EncodeLocalName(member.Name);
-                            object[] optionalFields = null;
-                            if (optionalFields == null || optionalFields.Length == 0)
```

Comment:
On desktop, this is set as (see the code [here](http://referencesource.microsoft.com/#System.Runtime.Serialization/System/Runtime/Serialization/ClassDataContract.cs,981)),
```c#
object[] optionalFields = field.GetCustomAttributes(Globals.TypeOfOptionalFieldAttribute, false);
```
As we are going to add `OptionalFieldAttribute` in Core, I think we should keep the cod here.